src/pages/index.tsx

```diff
@@ -28,15 +28,43 @@ const HomePage: NextPageWithLayout = () => {
   const [componentProps, setComponentProps] = useState<Record<string, unknown>>({});
   const [showNotificationModalState, setShowNotificationModalState] = useState(false);
   const accountId = useAuthStore((store) => store.accountId);
+  const [tosData, setTosData] = useState<TosData | null>(null);
+  const cacheTosData = useMemo(() => tosData, [tosData?.latestTosVersion]);
+
+  const handleModalCloseOnEsc = useCallback(() => {
+    setShowNotificationModalState(false);
+  }, []);
+
+  const turnNotificationsOn = useCallback(() => handleTurnOn(accountId, () => {
+    setShowNotificationModalState(false);
+  }), []);
+
+  const pauseNotifications = useCallback(() => {
+    handleOnCancel();
+    setShowNotificationModalState(false);
+  }, []);
+
+  const checkNotificationModal = useCallback(() => {
+    if (tosData && tosData.agreementsForUser.length > 0) {
+      // show notification modal for new users
+      const tosAccepted = tosData.agreementsForUser[tosData.agreementsForUser.length - 1].value === tosData.latestTosVersion;
+      // check if user has already turned on notifications
+      const { permission } = getNotificationLocalStorage();
```

Comment:
Is this a right prop to check does user already turned notifications on?

cc @marcinbodnar